### PR TITLE
Automatically cache etcd releases on GCS, to avoid triggering 403s

### PR DESCRIPTION
**Pull request by rmmh**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32171)

### CODE REVIEW

**Comment by rmmh** [comment]
Another solution is in: kubernetes/test-infra#522

This minimizes the effort for updating etcd-- just change ETCD_VERSION_DEFAULT and ETCD_DEFAULT_HASH, and whatever CI job has access to the kubernetes-jenkins bucket will update the cache.

**Comment by rmmh** [comment]
This might be a good cherrypick candidate, since it involves flake mitigation.

File: hack/lib/etcd.sh [hunk 1]
```diff
@@ -16,10 +16,14 @@
 
 # A set of helpers for starting/running etcd for tests
 
-ETCD_VERSION=${ETCD_VERSION:-3.0.4}
+ETCD_VERSION_DEFAULT=3.0.4
+ETCD_VERSION=${ETCD_VERSION:-$ETCD_VERSION_DEFAULT}
 ETCD_HOST=${ETCD_HOST:-127.0.0.1}
 ETCD_PORT=${ETCD_PORT:-2379}
 
+# This is the SHA256 digest of the linux-amd64 release tarball.
```
Comment (ixdy) [inline hack/lib/etcd.sh]: ew

Comment (rmmh) [inline hack/lib/etcd.sh]: This is definitely my least favorite part. Do you have any ideas on how to make this less gross, or better commented? I tried putting it in an associative array (keyed off the ETCD version), but that doesn't work on OSX bash, and it just got worse.

The digest is used to ensure that external resources are what we expect.

**Comment by rmmh** [comment]
Closing this because we haven't see the flake again.

**Comment by rmmh** [comment]
@apelisse this will fix some of the current bleeding

**Comment by rmmh** [comment]
@k8s-bot test this

**Comment by ixdy** [comment]
Is this still worth doing?

**Pull request closed without merging**